app/routes.py

```diff
@@ -1,2 +1,272 @@
+            'text': f'Someone just completed the task {task_to_update.title}'
+            } 
+        response = requests.post('https://slack.com/api/chat.postMessage', headers=headers, params=params)
+        return response
+####################### PATCH MARK INCOMPLETED - CRUD - UPDATE #############################
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"]) 
+def mark_incomplete(task_id):
+    task_to_update = Task.query.get(task_id)
+    if task_to_update is None:
+        return jsonify(None), 404
+    else:
+        task_to_update.completed_at = None
+
+        db.session.commit()
+            
+        jsonable_status_updated = task_to_update.to_dictionary()
+
+        return jsonify(task=jsonable_status_updated), 200
+
+
+####################### POST GOAL CRUD - CREATE #############################
+@goals_bp.route("", methods=["POST"]) 
+def create_goal(): 
+    request_body = request.get_json()
+    if 'title' not in request_body.keys(): # this condition works withour the keys() function too
+        response_bod = {"details": "Invalid data"}
+        return jsonify(response_bod), 400
+    else:
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal) # adds the goal model to the database
+        db.session.commit() # saves the goal to the database
+        
+        jsonable_new_goal = new_goal.to_dictionary()
+
+        return jsonify(goal=jsonable_new_goal), 201
+
+
+####################### DELETE GOAL - CRUD - DELETE #############################
+######### MUST HAVE A GET GOAL WORKING BEFORE DELETE CAN WORK ########
+@goals_bp.route("/<goal_id>", methods=["DELETE"]) 
+def delete_goal(goal_id):
+    delete_goal = Goal.query.get(goal_id)
+    if delete_goal is None:
+        return jsonify(None), 404
+    else:
+        db.session.delete(delete_goal) # deletes the model from the database
+        db.session.commit() # Save Action
+    
+        response_body = f"Goal {goal_id} \"{delete_goal.title}\" successfully deleted"
+        
+        return jsonify(details=response_body), 200
+
+
+####################### GET ALL GOALS - CRUD - READ #######################################
+@goals_bp.route("", methods=["GET"]) 
+def get_all_goals():
+    goals = Goal.query.all()
+
+    goals_response = []
+    for g in goals: 
+        each_goal = g.to_dictionary()
+        goals_response.append(each_goal)
+    
+    return jsonify(goals_response), 200
+    
+    # return render_template("index.html", goals_response=goals_response)
+
+####################### GET ONE GOAL by ID - CRUD - READ #############################
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        response = {
+            "id": goal.id,
+            "title": goal.title
+        }
+        return jsonify(goal=goal.to_dictionary()), 200
+    else:
+        response = None
+        return jsonify(response), 404
```

Comment:
It seems like you were trying to create a response object and then pass it to jsonify.  Consider something like this:

```suggestion
    if goal:
        response = {
           "goal":goal.to_dictionary()
        }
        code = 200
    else:
        response = None
        code = 404
    return jsonify(response), code
```